sdk/core/core-http/lib/proxyAgent.ts

```diff
@@ -10,16 +10,26 @@ import { URLBuilder } from "./url";
 import { HttpHeaders } from "./httpHeaders";
 
 export type ProxyAgent = { isHttps: boolean; agent: http.Agent | https.Agent };
-export function createProxyAgent(requestUrl: string, proxySettings: ProxySettings, headers?: HttpHeaders): ProxyAgent {
+
+export function createProxyAgent(
+  requestUrl: string,
+  proxySettings: ProxySettings,
+  headers?: HttpHeaders
+): ProxyAgent {
   const tunnelOptions: tunnel.HttpsOverHttpsOptions = {
     proxy: {
       host: URLBuilder.parse(proxySettings.host).getHost() as string,
       port: proxySettings.port,
-      headers: (headers && headers.rawHeaders()) || {}
+      headers: (headers && headers.rawHeaders()) || proxySettings.headers || {},
```

Comment:
Forgive me if I don't fully understand how proxy setting headers work.  This line sets `headers` to `proxySettings.headers` only when headers don't have raw headers. Is this intended? Or should we combine `proxySettings.headers` with `headers.rawHeaders()`?

Reply:
Agree, I wanted to discuss about it.
We are currently passing the request headers using the method parameter (and this works)
Any additional proxy specific headers that user wishes to pass should likely be combined with it.

Reply:
It does look like we should be combining the two

Reply:
Updated